implementations/rust/ockam/ockam_app/src/app/app_state.rs

```diff
@@ -1,10 +1,11 @@
-use std::sync::Arc;
+use std::sync::{Arc, RwLock};
```

Comment:
We'll need to be mindful about not holding locks across an `await`, but I think using a tokio type instead would include a different set of footguns footguns.

Reply:
Yes the whole topic is a can of worms (I read [this interesting post](https://cliffle.com/blog/lilos-cancel-safety/) about it btw).